include/seqan3/range/views/pairwise_combine.hpp

```diff
@@ -639,7 +252,403 @@ class pairwise_combine_view : public std::ranges::view_interface<pairwise_combin
+ * iterator points to the end of the underlying range and the first iterator to the last element of the range.
+ * Also note that this iterator does not model [Cpp17Iterator](https://en.cppreference.com/w/cpp/named_req/Iterator),
+ * since it does not return a reference to the represented type but a prvalue.
+ * Thus this iterator might not be usable with some legacy algorithms of the STL. But it is guaranteed to work with
+ * the ranges algorithms.
+ */
+ template <std::ranges::view underlying_range_type>
+ //!\cond
+     requires std::ranges::forward_range<underlying_range_type> && std::ranges::common_range<underlying_range_type>
+ // !\endcond
+template <typename range_type>
+class pairwise_combine_view<underlying_range_type>::iterator_type :
+                                    public std::ranges::view_interface<pairwise_combine_view<underlying_range_type>>
+{
+private:
+
+    //!\brief Friend declaration for iterator with different range const-ness.
+    template <typename other_range_type>
+        requires std::same_as<std::remove_const_t<range_type>, std::remove_const_t<other_range_type>>
+    friend class iterator_type;
+
+    //!\brief Alias type for the iterator over the passed range type.
+    using underlying_iterator_type = std::ranges::iterator_t<range_type>;
+    //!\brief Alias for the value type of the underlying iterator type.
+    using underlying_val_t = typename std::iterator_traits<underlying_iterator_type>::value_type;
+    //!\brief Alias for the reference type of the underlying iterator type.
+    using underlying_ref_t = typename std::iterator_traits<underlying_iterator_type>::reference;
+
+public:
+    /*!\name Associated types
+     * \{
+     */
+
+    //!\brief The difference type.
+    using difference_type   = std::ptrdiff_t;
+    //!\brief The value type.
+    using value_type        = std::tuple<underlying_val_t, underlying_val_t>;
+    //!\brief The reference type.
+    using reference         = common_tuple<underlying_ref_t, underlying_ref_t>;
+    //!\brief The pointer type.
+    using pointer           = void;
+    //!\brief The iterator category tag.
+    using iterator_category = iterator_tag_t<underlying_iterator_type>;
+    //!\}
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr iterator_type()                                  noexcept = default; //!< Defaulted.
+    constexpr iterator_type(iterator_type const &)             noexcept = default; //!< Defaulted.
+    constexpr iterator_type(iterator_type &&)                  noexcept = default; //!< Defaulted.
+    constexpr iterator_type & operator=(iterator_type const &) noexcept = default; //!< Defaulted.
+    constexpr iterator_type & operator=(iterator_type &&)      noexcept = default; //!< Defaulted.
+    ~iterator_type()                                           noexcept = default; //!< Defaulted.
```

Comment:
```suggestion
    iterator_type() = default; //!< Defaulted.
    iterator_type(iterator_type const &) = default; //!< Defaulted.
    iterator_type(iterator_type &&) = default; //!< Defaulted.
    iterator_type & operator=(iterator_type const  = default; //!< Defaulted.
    iterator_type & operator=(iterator_type &&) = default; //!< Defaulted.
    ~iterator_type() = default; //!< Defaulted.
```

for explicitly defaulted constructors and assignment operators, `constexpr` and `noexcept` is automatically added :)

Reply:
Thanks ! So it is redundant but harmless ? :)

Reply:
Yes